packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthCognito.kt

```diff
@@ -364,8 +364,22 @@ public class AuthCognito : FlutterPlugin, ActivityAware, MethodCallHandler {
                 if (req.getAWSCredentials) {
                   val cognitoAuthSession = result as AWSCognitoAuthSession
                   when (cognitoAuthSession.identityId.type) {
-                    AuthSessionResult.Type.SUCCESS -> prepareCognitoSessionResult(flutterResult, cognitoAuthSession)
-                    AuthSessionResult.Type.FAILURE -> prepareCognitoSessionFailure(flutterResult, cognitoAuthSession)
+                    AuthSessionResult.Type.SUCCESS -> {
+                      prepareCognitoSessionResult(flutterResult, cognitoAuthSession)
+                    }
+                    AuthSessionResult.Type.FAILURE -> {
+                      // checking case where user pool data is available but id pool does not exist
+                      if (
+                            result.identityId.error is AuthException.InvalidAccountTypeException &&
+                            result.awsCredentials.error is AuthException.InvalidAccountTypeException &&
+                            result.userPoolTokens.type.toString() == "SUCCESS" &&
+                            result.userSub.type.toString() == "SUCCESS"
```

Comment:
Is it NPE safe to access all these properties in an event of `AuthSessionResult.Type.FAILURE`

Reply:
@jamesonwilliams ^^

Reply:
I'm not sure; let me check the source. Do you get nullability guides in Android Studio? To be safe, you can use Kotlin's ["safe call operator"](https://kotlinlang.org/docs/reference/null-safety.html#safe-calls). In the worst case, the following should evaluate to `null == "SUCCESS"`, as opposed to throwing a `NullPointerException`:
```kotlin
result?.userSub?.type?.toString() == "SUCCESS"
```

Reply:
Yea, as far as I can tell, all of your current calls are safe. All of the children of `result` are non-null. The `type` field is non-null, so `toString()` is okay. `error` _might be null_, but that's okay, since it's at the end of your clauses.